csharp/ql/src/semmle/code/csharp/security/dataflow/TaintedPath.qll

```diff
@@ -38,6 +43,10 @@ module TaintedPath {
     override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
 
     override predicate isSanitizer(DataFlow::Node node) { node instanceof Sanitizer }
+
+    override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
+      guard instanceof BarrierGuard
```

Comment:
I think this is confusing. I read it as if you were adding all `BarrierGuards` until I saw that you had defined another abstract class with the same name.

Reply:
Yeah, but this is the general pattern in C# queries, for extensibility.

Reply:
The difference being that `BarrierGuard` is an already existing class.  Having `abstract class Sink extends DataFlow::Node` is fine as is quite clear that there's an important distinction between `Sink` and `Node`, but doing `abstract class BarrierGuard extends DataFlow::BarrierGuard` is confusing since then you suddenly have a very important distinction between two classes with identical names.

Reply:
(I think my comment was lost, it was: `Yeah, but this is the general pattern in C# queries, for extensibility`.)